docs/docs/concepts/memory.md

```diff
@@ -132,44 +155,124 @@ def summarize_conversation(state: State):
 
-## Few Shot Examples
+## Long term memory
 
-Few-shot learning is a powerful technique where LLMs can be ["programmed"](https://x.com/karpathy/status/1627366413840322562) inside the prompt with input-output examples to perform diverse tasks. While various [best-practices](https://python.langchain.com/docs/concepts/#1-generating-examples) can be used to generate few-shot examples, often the challenge lies in selecting the most relevant examples based on user input. 
+Long-term memory refers to the ability of a system to remember information between different conversations (or interactions).
+There a many different ways this can be accomplished, and the right way to do so depends entirely on your application.
+Rather than build a generic "long-term memory" solution, LangGraph aims to give you the ability to more directly control the long-term memory of your application.
+You can use LangGraph's [Store](persistence.md#memory-store) to accomplish this.
 
-LangChain [`ExampleSelectors`](https://python.langchain.com/docs/how_to/#example-selectors) can be used to customize few-shot example selection from a collection of examples using criteria such as length, semantic similarity, semantic ngram overlap, or maximal marginal relevance.
+Long-term memory is far from a solved problem.
+It is hard to provide generic advice on how to build long term memory, as it often depends on your application.
+Still, there are several patterns you should be thinking about when implementing long-term memory, and we try outline those below.
 
-If few-shot examples are stored in a [LangSmith Dataset](https://docs.smith.langchain.com/how_to_guides/datasets), then dynamic few-shot example selectors can be used out-of-the box to achieve this same goal. LangSmith will index the dataset for you and enable retrieval of few shot examples that are most relevant to the user input based upon keyword similarity ([using a BM25-like algorithm](https://docs.smith.langchain.com/how_to_guides/datasets/index_datasets_for_dynamic_few_shot_example_selection) for keyword based similarity). 
+**Do you want to update memory "in the hot path" or "as a separate process"**
 
-See this how-to [video](https://www.youtube.com/watch?v=37VaU7e7t5o) for example usage of dynamic few-shot example selection in LangSmith. Also, see this [blog post](https://blog.langchain.dev/few-shot-prompting-to-improve-tool-calling-performance/) showcasing few-shot prompting to improve tool calling performance and this [blog post](https://blog.langchain.dev/aligning-llm-as-a-judge-with-human-preferences/) using few-shot example to align an LLMs to human preferences.
+Memory can be updated either as part of the application logic (e.g. "in the hot path" of the application) or as a separate process.
```

Comment:
Why do we call it "separate process" vs. "in parallel" (vs. in the background)

Is the fact that it's a process vs. a thread important or just an implementation detail?

Reply:
it's actually a sep thread not process; mainly conceptually

Reply:
no, the difference is that it's either part of the app or a call to update from outside of the app logic, worth clarifying this language